plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -184,21 +184,30 @@ fu_plugin_thunderbolt_add (FuPlugin *plugin, GUdevDevice *device)
 	}
 
 	vid = fu_plugin_thunderbolt_udev_get_id (device, "vendor", &error);
-	if (vid == 0x0) {
+	if (vid == 0x0)
 		g_warning ("failed to get Vendor ID: %s", error->message);
-		return;
-	}
+
 	did = fu_plugin_thunderbolt_udev_get_id (device, "device", &error);
-	if (did == 0x0) {
+	if (did == 0x0)
 		g_warning ("failed to get Device ID: %s", error->message);
-		return;
-	}
 
 	is_host = fu_plugin_thunderbolt_is_host (device);
-	vendor_id = g_strdup_printf ("TBT:0x%04X", (guint) vid);
-	device_id = g_strdup_printf ("TBT-%04x%04x", (guint) vid, (guint) did);
+
+	version = g_udev_device_get_sysfs_attr (device, "nvm_version");
+	if (is_host && ((version == NULL) || (vid == 0x0) || (did == 0x0))) {
```

Comment:
So, neither `g_file_get_contents` nor `g_file_load_contents` will work, because in both cases the error `ENODATA` are not correctly mapped (e.g. see [here](https://github.com/GNOME/glib/blob/master/gio/gioerror.c#L70) and [here](https://github.com/GNOME/glib/blob/master/glib/gfileutils.c#L485)). I think we have to fall back to `read`. Unless we just assume `nvm_version == NULL` means safe mode.